torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2231,14 +2523,16 @@ def _add_guard(self, expr: "sympy.Expr") -> None:
         if torch._dynamo.config.print_guards:
             if log.level <= logging.WARNING:
                 # reusing flag that prints guards
-                frame_summaries = TracingContext.get().frame_summary_stack
-                # frame_summaries describes a stack of functions
-                # TODO(avik): It would be better to describe a stack of function calls instead
-                current_loc = TracingContext.get().loc_in_frame
-                # current_loc describes a line in the current frame
-                user_stack = ''.join(traceback.format_list([*frame_summaries, current_loc]))
-                expr = LoggingShapeGuardPrinter(self.var_to_sources).doprint(expr)
-                log.warning(f"Adding shape guard {expr} at \n{user_stack}")
+                tc = TracingContext.get()
+                if tc is not None:
```

Comment:
Wait why this change? TC should never be None here.

Reply:
Cut. I remember running into a problem here, maybe in some other context...